cms/djangoapps/contentstore/asset_storage_handlers.py

```diff
@@ -88,6 +88,50 @@ def handle_assets(request, course_key_string=None, asset_key_string=None):
     return HttpResponseNotFound()
 
 
+def check_asset_pre_upload(request, course_key_string):
+    """
+    Check if the files have any filename conflicts with existing assets.
+
+    Returns a JSON response with the following fields:
+    - allowed: boolean, whether the user is allowed to upload an asset
+    - reason: string, the reason why the user is not allowed to upload an asset
+    """
+    course_key = CourseKey.from_string(course_key_string)
+    result = _get_pre_existing_filenames(request, course_key)
+    if len(result) > 0:
+        return JsonResponse({
+            'allowed': False,
+            'reason': _('The following files already exist: {}').format(', '.join(result))
+        }, status=409)
+    else:
+        return JsonResponse({
+            'allowed': True,
+            'reason': ''
+        })
+
+
+def _get_pre_existing_filenames(request, course_key):
+    """
+    Get a list of filenames that already exist in the course.
+    """
+    if not has_course_author_access(request.user, course_key):
```

Comment:
@mavidser Can we move this check to the start of `check_asset_pre_upload` function?